src/pages/playground/Playground.tsx

```diff
@@ -346,9 +440,50 @@ const Playground: React.FC<PlaygroundProps> = props => {
   }, [editorSessionId]);
 
   const hash = isSicpEditor ? props.initialEditorValueHash : location.hash;
+  const { uuid } = useParams<string>();
+
+  const handleURL = useCallback(
+    (uuid: string | undefined) => {
+      if (uuid !== undefined) {
+        fetch(`${Constants.backendUrl}/api/shared_programs/${uuid}`)
+          .then(response => response.json())
+          .then(resp => {
+            const res: ShareLinkState = new ShareLinkStateDecoder(resp).decodeWith(
+              new JsonDecoderDelegate()
+            );
+            resetConfig(
+              res,
+              { handleChangeExecTime, handleChapterSelect },
+              workspaceLocation,
+              dispatch,
+              fileSystem
+            );
+          });
```

Comment:
Ditto on using our request helper instead of `fetch` directly.

Reply:
have changed to `request`  in `RequestHelper.tsx`. Not sure if I'm using correctly here, please check the latest update for details. Thanks!